### PR TITLE
fix(): Refactor code for better readability and adherence to best pra…

### PR DESCRIPTION
**Pull request by EmekaManuel**
- Replaced wildcard imports with specific imports for improved efficiency
- Added semicolons and ensured consistent formatting across the codebase
- Removed unnecessary IIFE wrappers and improved error handling
- Updated the initializeSignerKeypair, airdropSolIfNeeded, and sendTestMovieReview functions for better clarity
- Adjusted the main function to follow modern async patterns

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -1,104 +1,122 @@
...
+import dotenv from "dotenv";
+
+dotenv.config();
+
+function initializeSignerKeypair(): Keypair {
+  if (!process.env.PRIVATE_KEY) {
+    console.log("Creating .env file");
+    const signer = Keypair.generate();
+    writeFileSync(".env", `PRIVATE_KEY=[${signer.secretKey.toString()}]`);
+    return signer;
```
Comment (mikemaccana) [inline src/index.ts]: Use https://github.com/solana-developers/helpers?tab=readme-ov-file#load-or-create-a-keypair-and-airdrop-to-it-if-needed

File: src/index.ts [hunk 2]
```diff
@@ -1,104 +1,122 @@
...
\ No newline at end of file
+main()
+  .then(() => {
+    console.log("Finished successfully");
+    process.exit(0);
+  })
+  .catch((error) => {
+    console.error(error);
+    process.exit(1);
+  });
```
Comment (mikemaccana) [inline src/index.ts]: Avoid `.then()` - we use `esrun` and it supports top level `await` (see CONTRIBUTING).

**Review by mikemaccana — CHANGES_REQUESTED** [review]
There's one of the helpers that should be used here, will save our audience a bunch of time, see the comment above. Also note re: top level await below.

File: src/index.ts (hunk 2 above)
Comment (EmekaManuel) [reply src/index.ts]: fixed ✅

File: src/index.ts (hunk 1 above)
Comment (EmekaManuel) [reply src/index.ts]: I implemented this but on running the code, there seems to be a persistent error being thrown from the package `@solana-developers/helpers`. View my implementation and the error being thrown below. 

Screenshot 2024-09-11 at 16 03 55
Screenshot 2024-09-11 at 16 04 22

Comment (mikemaccana) [reply src/index.ts]: We fixed an error in that module today - can you try https://www.npmjs.com/package/@solana-developers/helpers 2.5.5 and let me know?

File: src/index.ts [hunk 3]
```diff
@@ -1,104 +1,119 @@
...
-    transaction.add(instruction)
-    const tx = await web3.sendAndConfirmTransaction(connection, transaction, [signer])
-    console.log(`https://explorer.solana.com/tx/${tx}?cluster=devnet`)
+async function airdropSolIfNeeded(signer: Keypair, connection: Connection) {
+  const balance = await connection.getBalance(signer.publicKey);
+  console.log("Current balance is", balance);
+  if (balance < LAMPORTS_PER_SOL) {
+    console.log("Airdropping 1 SOL...");
+    await connection.requestAirdrop(signer.publicKey, LAMPORTS_PER_SOL);
+  }
```
Comment (mikemaccana) [inline src/index.ts]: Just use the helper. If the helper is broken, let's fix the helper rather than making a second copy of the same code.

**Review by mikemaccana — CHANGES_REQUESTED** [review]
See comments above.

File: src/index.ts (hunk 3 above)
Comment (EmekaManuel) [reply src/index.ts]: Implemented ✅

File: src/index.ts (hunk 1 above)
Comment (EmekaManuel) [reply src/index.ts]: Hello @mikemaccana  the new version worked perfectly now. Great job on the fix!

Screenshot 2024-09-12 at 22 31 49

File: src/index.ts [hunk 4]
```diff
@@ -12,30 +12,34 @@ import {
...
+// }
+
+// async function airdropSolIfNeeded(signer: Keypair, connection: Connection) {
+//   const balance = await connection.getBalance(signer.publicKey);
+//   console.log("Current balance is", balance);
+//   if (balance < LAMPORTS_PER_SOL) {
+//     console.log("Airdropping 1 SOL...");
+//     await connection.requestAirdrop(signer.publicKey, LAMPORTS_PER_SOL);
+//   }
+// }
```
Comment (mikemaccana) [inline src/index.ts]: Remove the dead code. See https://stackoverflow.com/questions/15699995/could-someone-explain-the-pros-of-deleting-or-keeping-unused-code

**Review by mikemaccana — CHANGES_REQUESTED** [review]
See comment below.

File: src/index.ts (hunk 4 above)
Comment (mikemaccana) [inline src/index.ts]:
```suggestion
```

**Pull request merged**